QueryManager/src/test/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryControllerTest.java

```diff
@@ -0,0 +1,220 @@
+				+ "                \"type\": \"Property\",\r\n" + "                \"value\": \"GPS_NEW\"\r\n"
+				+ "            },\r\n" + "            \"value\": 52.5\r\n" + "        }\r\n" + "    ],\r\n"
+				+ "    \"location\": {\r\n" + "        \"type\": \"GeoProperty\",\r\n" + "        \"value\": {\r\n"
+				+ "            \"type\": \"Point\",\r\n" + "            \"coordinates\": [\r\n"
+				+ "                -8.5,\r\n" + "                41.2\r\n" + "            ]\r\n" + "        }\r\n"
+				+ "    }\r\n" + "}"));
+
+	}
+
+	@AfterEach
+	public void tearDown() {
+		entity = null;
+		response = null;
+		entities = null;
+	}
+
+	/**
+	 * this method is use for get Entity By Id Not Found
+	 */
+	@Test
+	@Order(1)
+	public void getEntityByIdNotFoundTest() throws Exception {
+		try {
+			ExtractableResponse<Response> response = given().accept(AppConstants.NGB_APPLICATION_JSONLD).request()
+					.contentType(AppConstants.NGB_APPLICATION_JSON).when()
+					.get("/ngsi-ld/v1/entities/urn:ngsi-ld:Vehicle:A102").then().statusCode(404).extract();
+			assertEquals(404, response.statusCode());
+		} catch (Exception e) {
+			Assertions.fail();
+			e.printStackTrace();
+		}
+
+	}
+
+	/**
+	 * this method is use for get Entites based on Quaery
+	 */
+	@Test
+	@Order(2)
+	public void getQueryForEntitiesTest() throws Exception {
+		try {
+			ExtractableResponse<Response> response = given().accept(AppConstants.NGB_APPLICATION_JSONLD).request()
+					.contentType(AppConstants.NGB_APPLICATION_JSON).when().get("/ngsi-ld/v1/entities?type=v").then()
+					.statusCode(Status.OK.getStatusCode()).statusCode(200).extract();
+			assertEquals(200, response.statusCode());
+		} catch (Exception e) {
+			Assertions.fail();
+			e.printStackTrace();
+		}
+
+	}
+
+	/**
+	 * this method is use for get Entites based on Quaery BadRequest
+	 */
+	@Test
+	@Order(3)
+	public void getQueryForEntitiesBadRequestTest() {
+		try {
+			ExtractableResponse<Response> response = given().accept(AppConstants.NGB_APPLICATION_JSONLD).request()
+					.contentType(AppConstants.NGB_APPLICATION_JSON).when().get("/ngsi-ld/v1/entities?type1=Test50231.1")
+					.then().statusCode(400).extract();
+			assertEquals(400, response.statusCode());
+		} catch (Exception e) {
+			Assertions.fail();
+			e.printStackTrace();
+		}
+
+	}
+
+	/**
+	 * this method is use for get All Entites based on Type
+	 */
+	@Test
+	@Order(4)
+	public void getAllTypesTest() throws Exception {
+		try {
+			ExtractableResponse<Response> response = given().accept(AppConstants.NGB_APPLICATION_JSONLD).request()
+					.contentType(AppConstants.NGB_APPLICATION_JSON).when().get("/ngsi-ld/v1/types").then()
+					.statusCode(Status.OK.getStatusCode()).statusCode(200).extract();
+			assertEquals(200, response.statusCode());
+		} catch (Exception e) {
+			Assertions.fail();
+			e.printStackTrace();
+		}
+
+	}
+
+	/**
+	 * this method is use for get Entites based on entity type
```

Comment:
Add complete comment "this method is use for get entities based on entity type if entity type is not exist."

Reply:
done